src/api/app/models/package.rb

```diff
@@ -859,6 +859,17 @@ def rev
     dir_hash['rev']
   end
 
+  def self.current_revision(project, package)
+    dirhash = Directory.hashed(project: project, package: package)
```

Comment:
seems this is not listing the content in expanded mode. I think this what is wanted at least for "current branch".